### DISCUSSION TITLE
Which modules are available

### DISCUSSION THREAD

**Discussion by YatishRam** [opensafely/documentation#156] [Q&A] [answered]
Does anyone know if custom python modules are available or only native modules are supported?
e.g pandas, csvkit

 If I were to do some csv processing, can I make use of theses libraries or do I have to use the native python code?

**Comment by evansd** [comment, marked as answer]
Hi @YatishRam, you can find the list of available packages in the repository we use to build our Python Docker image (we should work on a way to automatically include this in the documentation though).
https://github.com/opensafely-core/python-docker/blob/main/requirements.in

The exact versions installed are recorded in this (slightly less readable) file:
https://github.com/opensafely-core/python-docker/blob/main/requirements.txt
